lib/src/arithmetic/fixed_point/fixed_point_logic.dart

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// fixed_point_logic.dart
+// Implementation of Fixed Point objects
+//
+// 2024 September 25
+// Author: Soner Yaldiz <soner.yaldiz@intel.com>
+
+import 'package:rohd/rohd.dart';
+// import 'package:rohd_hcl/src/arithmetic/fixed_point/fixed_point_value.dart';
+import 'package:rohd_hcl/src/exceptions.dart';
+
+/// Fixed point logic representation
+class FixedPoint extends Logic {
```

Comment:
`FixedPoint` is simply extending `Logic` as fixed point arithmetic in two's complement format re-uses integer arithmetic. @desmonddak , is this okay?

Reply:
The only challenge I could see is that you need extra state to store the 'dot' location and so you could run into some challenge with returning FixedPoint as an output (as we cast unsafely to do this -- see the FloatingPoint structure and the  Modules that manipulate.

What happens in FloatingPoint is that it extends LogicStructure because of the different fields, even though you could think of it as a single Logic.  I'm not sure how you are going to store the 'state' of the 'dot'.  Definitely a Max question on the starting point here.

Reply:
For 8-bit floating point to fixed point conversion, this data type is not used as the output format varies according to FP8 format. Therefore, this data type can be removed from this PR and revisited in the future.

Reply:
i think this looks great!  any operations which "know" about fixed point can properly return a fixedpoint signal with proper population.